src/track/beatgrid.cpp

```diff
@@ -184,13 +190,49 @@ double BeatGrid::findNthBeat(double dSamples, int n) const {
         n = n + 1;
     }
 
-    double dResult = dClosestBeat + n * m_dBeatLength;
+    double dResult = floor(dClosestBeat + n * m_dBeatLength);
     if (!even(static_cast<int>(dResult))) {
         dResult--;
     }
     return dResult;
 }
 
+QPair<double, double> BeatGrid::findPrevNextBeats(double dSamples) const {
+    QMutexLocker locker(&m_mutex);
```

Comment:
Nope, you can mutate a beat object unfortunately (beats_translate, double / halve BPM / etc. all mutate the beats object in place). We could get rid of the locks if they were immutable.